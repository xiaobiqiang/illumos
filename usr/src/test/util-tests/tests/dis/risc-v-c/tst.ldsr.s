/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test common compact loads and stores.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	c.lwsp	s1,(sp)
	c.lwsp	a4,0x40(sp)
	c.lwsp	t3,0x34(sp)
	c.swsp	t4,(sp)
	c.swsp	tp,0x20(sp)
	/*
	 * gas 2.30 doesn't support using the ABI aliases. However, that's how
	 * we disassemble these.
	 */
	c.lw	x9,(x10)
	c.lw	x12,0x40(x13)
	c.fld	fs1,(a0)
	c.fld	fa3,0x40(a3)
.size libdis_test, [.-libdis_test]
