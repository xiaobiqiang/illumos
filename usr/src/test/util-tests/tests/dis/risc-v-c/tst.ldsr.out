    libdis_test:      82 44              c.lwsp s1,0x0(sp)
    libdis_test+0x2:  06 47              c.lwsp a4,0x40(sp)
    libdis_test+0x4:  52 5e              c.lwsp t3,0x34(sp)
    libdis_test+0x6:  76 c0              c.swsp t4,0x0(sp)
    libdis_test+0x8:  12 d0              c.swsp tp,0x20(sp)
    libdis_test+0xa:  04 41              c.lw s1,0x0(a0)
    libdis_test+0xc:  b0 42              c.lw a2,0x40(a3)
    libdis_test+0xe:  04 21              c.fld fs1,0x0(a0)
    libdis_test+0x10: b4 22              c.fld fa3,0x40(a3)
